src/components/DatePicker/Day.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import styled from '@emotion/styled'
+import { css } from '@emotion/react'
+import { color, radius, fontWeight, device } from '../../theme'
+import { isSameDate } from '../../utils/dates'
+
+interface DayLabelProps {
+  isCurrentDay: boolean
+}
+
+interface DayProps {
+  date: Date
+  isSelected: boolean
+  isDisabled?: boolean
+  onSelect: (date: Date) => void
+}
+
+interface DayButtonProps {
+  selected: boolean
+  disabled?: boolean
+  onClick: () => void
+}
+
+const DayButton = styled.button<DayButtonProps>`
+  width: 46px;
+  height: 46px;
+  background-color: ${color.stardust};
+  border-radius: ${radius.sm};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  @media ${device.tablet} {
+    width: 54px;
+    height: 54px;
+  }
+
+  ${({ selected }) =>
+    selected &&
+    css`
+      background-color: ${color.earth};
+
+      p {
+        color: ${color.nova};
+
+        [data-theme='dark'] & {
+          color: ${color.space};
+        }
+      }
+    `}
+
+  ${({ disabled }) =>
+    disabled &&
+    css`
+      opacity: 0.5;
+      cursor: unset;
+    `}
+`
+
+const DayLabel = styled.p<DayLabelProps>`
+  font-weight: ${props =>
+    props.isCurrentDay ? fontWeight.semiBold : fontWeight.regular};
+  color: ${color.space};
+`
+
+const Day = ({ date, isSelected, isDisabled, onSelect }: DayProps) => {
+  const isCurrentDay = isSameDate(date, new Date())
+
+  const handleSelect = () => {
+    if (!isDisabled) {
+      onSelect(date)
+    }
+  }
+
+  return (
+    <DayButton
+      selected={isSelected}
+      aria-selected={isSelected}
+      {...(isDisabled ? { disabled: true } : {})}
```

Comment:
This can just be a boolean with either true or false right? Otherwise the type should be `disabled?: true` instead of `disabled?: boolean` 🤔

Reply:
Actually, this is not needed since it will inherit the prop from the HTML button type. thanks for noticing  🙏